Tutorials/Style/Functions-Docstrings.MD

```diff
@@ -0,0 +1,169 @@
+    code that turns your inputs 
+    into something you want to output
+    return output
+```
+
+### Understanding unfamiliar functions
+
+Often in scientific work, there are functions that other people have written and are available to us to use in our own code, but we don't know how to implement them (for instance,  mathematical functions in `numpy` and `scipy` modules or plotting functions in the `matplotlib` module). Fortunately, python has a built in way to get help on unfamiliar functions: the `help()` function!
+
+For instance, take the built in function `len`. If we don't know what this function does, we can simply put the name of the function into the `help()` function and get some useful information about it. Running
+
+```python
+help(len)
+```
+
+returns
+
+```
+Help on built-in function len in module builtins:
+
+len(obj, /)
+    Return the number of items in a container.
+```
+
+This tells us that `len` is a built in function that comes with python, and that it takes one input (an object) and returns the number of items in that container object. 
+
+`help()` also works on other objects, such as variables, which can be useful if you aren't sure what a particular variable is. For instance:
+
+```python
+x = [1, 2]
+help(x)
+```
+
+tells us that `x` is a list, and gives us all sorts of information about the sorts of methods and functions that can be used with lists. 
+
+___
+
+## Docstrings
+
+What happens if you write a function for your own code and then make it publicly available for other people to use? How will users know what your function does if they need to understand it or adapt it? Well, you could comment your code (always good practice), but for functions (and classes) in particular, you can use the magic of **_docstrings_**. 
+
+### How do I write a docstring?
+
+It's easy to write a docstring. Just under your function definition, add triple quotes ` ''' `, write your docstring, and then end the docstring with another set of triple quotes ` ''' `. For example: 
+
+```python
+def square(x):
+    '''This is my docstring'''
+    return x**2
+```
+
+Now that we have our docstring, it will show up when we run `help()` on our function! This is a major advantage of using docstrings instead of comments when documenting functions. For instance: 
+
+```python
+def square(x):
+    '''This is my docstring'''
+    return x**2
+
+help(square)
+```
+
+returns
+
+```
+Help on function square in module __main__:
+
+square(x)
+    This is my docstring
+
+```
+
+which tells us that the function `square` is defined in the main code, it takes one input `x`, and that... it has a docstring that says `this is my docstring`. That's a docstring alright, but it isn't a very helpful one...
+
+### How do I write a _good_ docstring?
+
+It is only slightly harder to write a good docstring than it is to write a docstring. Best practices for writing a docstrings are: 
+- Describe what the function _does_ (this is usually the first line)
```

Comment:
A common style is to describe what the function returns rather than just what it does, i..e. `Return square of x`. This way the reader knows what the expected return value is as soon as they read the first line